packages/token-bridge-sdk/src/hooks/useTransactions.ts

```diff
@@ -521,6 +531,22 @@ const useTransactions = (): [Transaction[], TransactionActions] => {
     }
   }
 
+  const setDepositsInStore = (newTransactions: Transaction[]) => {
+    // appends the state with a new set of transactions
+    // useful when you want to display some transactions fetched from subgraph without worrying about existing state
+
+    let transactionsMap: { [id: string]: Transaction } = {}
+
+    ;[...transactions, ...newTransactions].forEach(tx => {
+      transactionsMap[tx.txID] = tx
+    })
+
+    return dispatch({
+      type: 'SET_TRANSACTIONS',
+      transactions: Object.values(transactionsMap)
+    })
+  }
```

Comment:
We only want to return the transaction right?
```
return dispatch({
  type: 'SET_TRANSACTIONS',
  transactions: [...transactions, ...newTransactions]
})
```

Reply:
It was done this way to remove duplicate entries between `transactions` and `newTransactions`.